src/DataProtection/DataProtection/src/Internal/HostingApplicationDiscriminator.cs

```diff
@@ -20,5 +21,6 @@ public HostingApplicationDiscriminator(IHostEnvironment hosting)
         _hosting = hosting;
     }
 
-    public string? Discriminator => _hosting?.ContentRootPath;
+    // Note: ContentRootPath behavior depends on the version, sometimes it has a trailing slash, we normalize by default by removing a trailing slash
+    public string? Discriminator => _hosting?.ContentRootPath?.TrimEnd(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar);
```

Comment:
Should it be stored in a field at construction instead of computed of every invocation?

Reply:
We don't know how IHostEnvironment implements ContentRootPath right?  Seems like the caching would belong there